apps/sms/js/activity_handler.js

```diff
@@ -518,28 +500,20 @@ var ActivityHandler = {
     // first to notify app that notification is clicked and then, once we show
     // Thread panel to the user, we remove that notification from the tray that
     // causes the second system message with "clicked" set to false.
-    if (!message.clicked) {
+    if (!notification.clicked) {
       // When app is run via notification system message there is no valid
       // current panel set hence app is in the invalid state, so let's fix this.
       Navigation.ensureCurrentPanel();
       return;
     }
 
-    navigator.mozApps.getSelf().onsuccess = function(event) {
-      var app = event.target.result;
-
-      app.launch();
+     navigator.mozApps.getSelf().then((app) => {
+       app.launch();
 
-      // the type param is only set for class0 messages
-      if (params.type === 'class0') {
```

Comment:
Looks like after the patch for the bug [1] we can't have notification for class-0 messages :)

[1] https://bugzilla.mozilla.org/show_bug.cgi?id=858908